phaser/builtin_steps.py

```diff
@@ -70,3 +76,73 @@ def filter_rows_step(batch, context, **kwargs):
                                     message=f"{num_dropped} rows dropped in filter_rows with '{func.__name__}'")
         return new_batch
     return filter_rows_step
+
+
+@row_step
+def flatten_all(row, context, **kwargs):
+    new_row = row.copy()
+    still_flattening = True
+    while still_flattening:
+        # Each iteration will pull nested data up one level. This will repeat until all nested data flattened.
+        still_flattening = False
+        for key, value in new_row.copy().items():
+            if isinstance(value, dict):
+                still_flattening = True
+                new_row, new_columns = _merge_values_if_no_collisions(new_row, key)
+    return new_row
+
+
+def _merge_values_if_no_collisions(row, key_name):
+    new_row = row.copy()
```

Comment:
This is the third copy of the row used in a single pass through `flatten_all`, and this `copy` is done for each dict in the row.  Are all the copy operations necessary for the logic to work and the data structure to not become corrupted?